### DISCUSSION TITLE
Add delay option for automatic actions when switching from AC to battery power for laptops using PD.

### DISCUSSION THREAD

**Discussion by tea40991103** [seerge/g-helper#2020] [Ideas]
Hi, I’m using ROG Flow X13 GV302XV with third-party PD power adapters and G-Helper. Most PD adapters with multiple ports will pause charging for 1 or 2 seconds when devices are connected/disconnected. This causes the laptop switching to battery for a few seconds and triggers G-Helper switching performance mode, GPU mode and screen refresh rate if they are set to automatic, causing unexpected exit for software using NVIDIA GPU and black screens when refresh rate changes. A delay timer and check power status again for automatic actions when switching from AC to battery power would perfectly solve the problem.

**Comment by seerge** [comment]
@tea40991103 try this build,

You need to add to the beginning of  ``%APPDATA%\GHelper\config.json`` following line (and save it) prior to launching app

```
"charger_delay" : 1000,
```

It will set a simple delay before processing power-supply change event in milliseconds (so 1000 = 1 second)

[GHelper.zip](https://github.com/seerge/g-helper/files/14151112/GHelper.zip)

**Reply by tea40991103** [reply]
@seerge Thank you for your super fast response. "charger_delay": 3000 works perfectly for my multi-port PD chargers :satisfied: I want to make a donation but Paypal said I'm regionally restricted :disappointed:

**Reply by seerge** [reply]
@tea40991103 oke, I will include this change  in next release then

**Reply by gnartS** [reply]
This is a great addition. A 3 second or so buffer prevents all the above when on USB-C (and other things like temporary power blips).

**Reply by PrinceOfHur** [reply]
Hi! I'm on a 2025, 5070ti. Love your software. My screen is frequently freezing when I go from battery to plugged in, forcing a hard shutdown to fix. I was wondering if the change was included in the newest release as referenced above, or do I need to add the charger_delay line to the config file as referenced above? I'm running Optimized GPU-mode, would setting it to ultimate make a difference?

**Reply by seerge** [reply]
@PrinceOfHur hello,

This change is long time live in the app, but don't forget that this an optional setting that you need to set in a config manually :)

But screen-freezing could be due to lots of different reasons. 
- Check if turning `Optimized` mode OFF makes difference (just set `Standard` for example)?
- Check if turning off `Auto` refresh rate makes difference

If it still happens after that, it could be just nvidia driver and it's own 'advanced optimus' acting. Set Nvidia Control Panel -> Display Mode specifically to `Optimus` setting or even try to do a clean Nvidia driver reinstall to the latest official driver from Nvidia website using DDU

**Reply by PrinceOfHur** [reply]
> @PrinceOfHur hello,
> 
> This change is long time live in the app, but don't forget that this an optional setting that you need to set in a config manually :)
> 
> But screen-freezing could be due to lots of different reasons.
> 
> * Check if turning `Optimized` mode OFF makes difference (just set `Standard` for example)?
> * Check if turning off `Auto` refresh rate makes difference
> 
> If it still happens after that, it could be just nvidia driver and it's own 'advanced optimus' acting. Set Nvidia Control Panel -> Display Mode specifically to `Optimus` setting or even try to do a clean Nvidia driver reinstall to the latest official driver from Nvidia website using DDU

Hello @seerge , thank you for continuing to support GHelper! Your recommendation of the charger delay to fix the plug-in freezes worked swimmingly; only 1 freeze event since i posted in 2025! Since the latest Ghelper update, i noticed the black screen transition after plug-in (while running) is taking an added 4-5 seconds to get back to desktop. I'm grateful it doesn't freeze, but what do you think is causing the delay? On reddit, i found another 2025 5070ti GA403WR user is experiencing the same thing since the last update. However, both of us performed the BIOS update too, so can't rule that out as the culprit either. Thoughts?

**Reply by seerge** [reply]
@PrinceOfHur that's just something beween your windows and GPU drivers. G-Helper has nothing to do with this. It just gives you a shortcut to existing refresh-rate change in windows Settings (and can automate toggle)

If screen goes black for 5 seconds if you change refresh rate in windows too - you have your  answer

**Reply by PrinceOfHur** [reply]
I manually changed the refresh rate, but the black screen lasts 1 second, 1.5 max. However, when I plugged back in the power cable just now, the black screen transition was just as smooth. I'll keep monitoring, but def not a problem.

**Comment by AudreyAP** [comment]
This is a really great addition! Thank you!

**Comment by thomaskeano** [comment]
I don't know where exactly to report this, but on the G14 GA402XY, the delay doesn't show.

![image](https://github.com/seerge/g-helper/assets/154497330/83975c31-e8f3-4a9b-ae10-75dfe157debb)

**Reply by seerge** [reply]
@thomaskeano its a parameter in a config. Literally read the thread above :)

**Reply by thomaskeano** [reply]
Misunderstood the update then! My bad haha. I thought it was a slider or a like an actual option, instead of a parameter in the config. Thank you for the response!

**Comment by versello** [comment]
This setting fixed my ROG Zephyrus G14 (2024) GA403 crashing when the laptop screen refresh rate was set to 'auto', and going from battery to plugged-in.
